xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp

```diff
@@ -50,14 +50,14 @@ using namespace XFILE::VIDEODATABASEDIRECTORY;
 //  Constructor is protected use ParseURL()
 CDirectoryNode::CDirectoryNode(NODE_TYPE Type, const std::string& strName, CDirectoryNode* pParent)
 {
-  m_Type=Type;
-  m_strName=strName;
-  m_pParent=pParent;
+  m_Type = Type;
+  m_strName = strName;
+  m_pParent = pParent;
 }
 
 CDirectoryNode::~CDirectoryNode()
 {
-  delete m_pParent;
+  delete m_pParent, m_pParent = nullptr;
```

Comment:
Like I mentioned, its coding habit and making sure ptrs are nulled after delete/free is the 1st thing I do when tracking down silly behavior. Nulling ptrs is pre-emptive paranoia from long years of coding :)

This very simple fix took days of debug, changes and leak testing to find and nail down and I had all sorts of nutty traps and triggers in the code that have been removed.  The null of the ptr can be removed but it does not really hurt and only adds a few lines of asm code. A purist might remove it as a nop but I tend to code on the defensive/paranoia side. You never know who might hold a bogus copy of the node and fetch it's parent. As long as the data for the code is still intact in memory, you would never see it fail and it would be a bitch to track down. This way, bang, instant crash.

Ultimately, your choice. MrMC will be keeping it.